src/start_env.rs

```diff
@@ -242,3 +296,48 @@ fn start_cnds(runtime: &mut Runtime, envfile: &mut EnvFile) {
         std::thread::sleep(std::time::Duration::from_millis(1000));
     }
 }
+
+fn handle_signal(
+    runtime: &mut Runtime,
+    envfile_path: PathBuf,
+    bitcoin_node: Arc<Node<BitcoinNode>>,
+    ethereum_node: Arc<Node<EthereumNode>>,
+) {
+    let terminate = Arc::new(AtomicBool::new(false));
+    signal_hook::flag::register(signal_hook::SIGTERM, Arc::clone(&terminate))
+        .expect("Could not register SIGTERM");
+    signal_hook::flag::register(signal_hook::SIGINT, Arc::clone(&terminate))
+        .expect("Could not register SIGINT");
+    signal_hook::flag::register(signal_hook::SIGQUIT, Arc::clone(&terminate))
+        .expect("Could not register SIGQUIT");
+    while !terminate.load(Ordering::Relaxed) {
+        sleep(Duration::from_millis(50))
+    }
+    println!("Signal received, terminating...");
+    clean_up(
+        runtime,
+        envfile_path,
+        Some(bitcoin_node),
+        Some(ethereum_node),
+    );
+}
+
+fn clean_up(
```

Comment:
Maybe split up this method `cleanup_bitcoin` and `cleanup_ethereum` as the arguments are orthogonal to each other. Then you can get rid of the `Option`.